classes/user/report/mappings/Notifications.inc.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * @defgroup lib_pkp_classes_user
+ */
+
+/**
+ * @file lib/pkp/classes/user/report/mappings/Notifications.inc.php
+ *
+ * Copyright (c) 2003-2020 Simon Fraser University
+ * Copyright (c) 2003-2020 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file LICENSE.
+ *
+ * @class Notifications
+ * @ingroup lib_pkp_classes_user
+ *
+ * @brief Retrieves a list of Mapping objects to display the user notifications.
+ */
+
+namespace PKP\User\Report\Mappings;
+use PKP\User\Report\{Report, Mapping, QueryBuilder, QueryListenerInterface, QueryOptions};
+use Illuminate\Database\Capsule\Manager as Capsule;
+
+class Notifications implements QueryListenerInterface {
+	/** @var string The key to extract the notification title */
+	private const NOTIFICATION_TITLE_KEY = 'settingKey';
+
+	/** @var string The setting which holds the notification status */
+	private const BLOCKED_NOTIFICATION_KEY = 'blocked_notification';
+
+	/** @var string The setting which holds the email notification status */
+	private const BLOCKED_EMAIL_NOTIFICATION_KEY = 'blocked_emailed_notification';
+
+	/** @var array A dictionary containing all notifications available in the application, the key is the notification ID */
+	private $_notificationsMap;
+
+	/**
+	 * Constructor
+	 * @param Report $report
+	 */
+	public function __construct(Report $report)
+	{
+		$this->_loadNotificationsMap();
+		$report->getQueryBuilder()->addListener($this);
+		$report->addMappings(...$this->_getMappings());
+	}
+
+	/**
+	 * @copydoc QueryListenerInterface::onQuery()
+	 */
+	public function onQuery(\Illuminate\Database\Query\Builder $query, QueryOptions $options): void
+	{
+		$query->leftJoin('notification_subscription_settings AS nss', function ($join) {
+			$join->on('nss.user_id', 'u.user_id')
+				->on('nss.context', 'ug.context_id');
+		});
+		switch (Capsule::connection()->getDriverName()) {
+			case 'mysql':
+				$options->columns[] = Capsule::raw("GROUP_CONCAT(DISTINCT CONCAT('[', nss.setting_name, '-', nss.setting_value, ']')) AS notifications");
+				break;
+			default:
+				$options->columns[] = Capsule::raw("STRING_AGG(DISTINCT CONCAT('[', nss.setting_name, '-', nss.setting_value, ']'), ',') AS notifications");
+				break;
+		}
```

Comment:
Is there something ready to add such exceptions in the code base? If not I'll probably create a `PKPCapsule extends Manager` to insert such custom methods, in case nobody has a better idea...